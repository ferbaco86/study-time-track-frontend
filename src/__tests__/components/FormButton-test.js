import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/store';
import FormButton from '../../components/FormButton';

describe('Rendering component', () => {
  it('creates a FormButton component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <FormButton>
              Add
            </FormButton>
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('Add');
  });
});

describe('FormButton', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <FormButton>
            Add
          </FormButton>
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
