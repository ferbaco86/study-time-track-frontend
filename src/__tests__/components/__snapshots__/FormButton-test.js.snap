// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormButton renders correctly 1`] = `
<button
  className="sc-bdfBwQ bJmxup"
>
  Add
</button>
`;
